packages/world/src/modules/core/implementations/WorldRegistrationSystem.sol

```diff
@@ -174,6 +174,19 @@ contract WorldRegistrationSystem is System, IWorldErrors {
     FunctionSelectors.set(worldFunctionSelector, resourceSelector, systemFunctionSelector);
   }
 
+  /**
+   * Register World function selectors in batch for the given namespace, name and system function.
+   */
+  function registerFunctionSelectors(
+    bytes32 resourceSelector,
+    string[] calldata systemFunctionNames,
+    string[] calldata systemFunctionArguments
+  ) public {
+    for (uint256 i = 0; i < systemFunctionNames.length; i++) {
+      registerFunctionSelector(resourceSelector, systemFunctionNames[i], systemFunctionArguments[i]);
+    }
+  }
```

Comment:
one small optimization we could do here is move the access control check outside this loop so we're not checking it on each iteration (via `registerFunctionSelector`)

Reply:
Reasonable. I have moved the access control check from the inner function to the batch function now.